virtcontainers/network.go

```diff
@@ -593,6 +593,33 @@ func initNetworkCommon(config NetworkConfig) (string, bool, error) {
 		return path, true, nil
 	}
 
+	// Check if the network namespace requested is the same as the runtime
+	var nsPath, currentNsPath string
+
+	// Here we are trying to resolve the path but it fails because
+	// namespaces links don't really exist. For this reason, the
+	// call to EvalSymlinks will fail when it will try to stat the
+	// resolved path found. As we only care about the path, we can
+	// retrieve it from the PathError structure.
```

Comment:
This looks good to me. I had exactly the same case some while ago, and I acknowledge this is the way to solve it.